opslevel/resource_opslevel_check_package_version.go

```diff
@@ -0,0 +1,314 @@
+	Name        types.String `tfsdk:"name"`
+	Notes       types.String `tfsdk:"notes"`
+	Owner       types.String `tfsdk:"owner"`
+
+	// TODO: Unique Fields Here
+	MissingPackageResult       types.String `tfsdk:"missing_package_result"`
+	PackageConstraint          types.String `tfsdk:"package_constraint"`
+	PackageManager             types.String `tfsdk:"package_manager"`
+	PackageName                types.String `tfsdk:"package_name"`
+	PackageNameIsRegex         types.Bool   `tfsdk:"package_name_is_regex"`
+	VersionConstraintPredicate types.Object `tfsdk:"version_constraint_predicate"`
+}
+
+func ParsePredicate(predicate *opslevel.Predicate) types.Object {
+	if predicate == nil {
+		return types.ObjectNull(predicateType)
+	}
+	predicateAttrValues := map[string]attr.Value{
+		"type":  types.StringValue(string(predicate.Type)),
+		"value": OptionalStringValue(predicate.Value),
+	}
+	return types.ObjectValueMust(predicateType, predicateAttrValues)
+}
+
+func NewCheckPackageVersionResourceModel(ctx context.Context, check opslevel.Check, planModel CheckPackageVersionResourceModel) CheckPackageVersionResourceModel {
+	var stateModel CheckPackageVersionResourceModel
+
+	stateModel.Category = RequiredStringValue(string(check.Category.Id))
+	if planModel.Enabled.IsNull() {
+		stateModel.Enabled = types.BoolValue(false)
+	} else {
+		stateModel.Enabled = OptionalBoolValue(&check.Enabled)
+	}
+	if planModel.EnableOn.IsNull() {
+		stateModel.EnableOn = types.StringNull()
+	} else {
+		// We pass through the plan value because of time formatting issue to ensure the state gets the exact value the customer specified
+		stateModel.EnableOn = planModel.EnableOn
+	}
+	stateModel.Filter = OptionalStringValue(string(check.Filter.Id))
+	stateModel.Id = ComputedStringValue(string(check.Id))
+	stateModel.Level = RequiredStringValue(string(check.Level.Id))
+	stateModel.Name = RequiredStringValue(check.Name)
+	stateModel.Notes = OptionalStringValue(check.Notes)
+	stateModel.Owner = OptionalStringValue(string(check.Owner.Team.Id))
+
+	if check.MissingPackageResult != nil {
+		stateModel.MissingPackageResult = RequiredStringValue(string(*check.MissingPackageResult))
+	}
+	stateModel.PackageConstraint = RequiredStringValue(string(check.PackageConstraint))
+	stateModel.PackageManager = RequiredStringValue(string(check.PackageManager))
+	stateModel.PackageName = RequiredStringValue(check.PackageName)
+	stateModel.PackageNameIsRegex = OptionalBoolValue(&check.PackageNameIsRegex)
+	stateModel.VersionConstraintPredicate = ParsePredicate(check.VersionConstraintPredicate)
+
+	return stateModel
+}
+
+func (r *CheckPackageVersionResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_check_package_version"
+}
+
+func (r *CheckPackageVersionResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Version: 1,
```

Comment:
```suggestion
```
I would leave this off since we aren't doing any StateUpgrader stuff here. By default it's `Version: 0` so if we were explicit it would need to be zero